File: Assignment38_5.c
#include <stdio.h>

///////////////////////////////////////////////////////////////////////////////
// Function Name : StrCpyToggle
// Description   : Toggle case while copying string
// Input         : String
// Output        : Modified string with toggled case
// Author        : Raj Samir Jadhav
// Date          : 02/12/2025
///////////////////////////////////////////////////////////////////////////////

void StrCpyToggle(char *src, char *dest)
{
    while(*src != '\0')                       // Traverse string
    {
        if((*src >= 'A') && (*src <= 'Z'))    // Capital → Small
        {
            *dest = *src + 32;
        }
        else if((*src >= 'a') && (*src <= 'z')) // Small → Capital
        {
            *dest = *src - 32;
        }
        else
        {
            *dest = *src;                       // Copy as it is
        }

        src++;
        dest++;
    }

    *dest = '\0';                                // NULL terminate destination
}

int main()
{
    char arr[50] = "Marvellous Python 2";
    char brr[50];                                  // Empty string

    StrCpyToggle(arr, brr);

    printf("%s\n", brr);  

    return 0;
}

File: Assignment38_5.java
import java.util.Scanner;

///////////////////////////////////////////////////////////////////////////////
// Function Name : StrCpyToggle
// Description   : Toggle case while copying string
// Input         : String
// Output        : Modified string with toggled case
// Author        : Raj Samir Jadhav
// Date          : 02/12/2025
///////////////////////////////////////////////////////////////////////////////
class Pattern
{
    public String StrCpyToggle(String src)
    {
        String dest = "";

        for(int i = 0; i < src.length(); i++)
        {
            char ch = src.charAt(i);

            if(ch >= 'A' && ch <= 'Z')        // Capital to Small
            {
                ch = (char)(ch + 32);
            }
            else if(ch >= 'a' && ch <= 'z')   // Small to Capital
            {
                ch = (char)(ch - 32);
            }

            dest = dest + ch;
        }
        return dest;
    }
}

class Assignment38_5
{
    public static void main(String[] args)
    {
        Scanner sobj = new Scanner(System.in);

        System.out.println("Enter string : ");
        String input = sobj.nextLine();

        Pattern pobj = new Pattern();
        String output = pobj.StrCpyToggle(input);

        System.out.println(output);

        sobj.close();
    }
}
